### ISSUE TITLE
[YCQL] Fatal 'Metadata is not yet loaded to this node'

### ISSUE THREAD

**Issue by tedyu** [yugabyte/yugabyte-db#5690]
User reported encountering the following crash on their 2.2.2 cluster:
```
F20200911 21:09:29 ../../src/yb/yql/cql/ql/ptree/pt_expr.h:1116] Check failed: desc_ Metadata is not yet loaded to this node
    @     0x7f868e459d64  yb::LogFatalHandlerSink::send()
    @     0x7f868d65eb36  google::LogMessage::SendToLog()
    @     0x7f868d65bcad  google::LogMessage::Flush()
    @     0x7f868d65ee89  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f869784eccb  yb::ql::PTRef::QLName()
    @     0x7f8697826a13  yb::ql::PTSelectStmt::AnalyzeOrderByClause()
    @     0x7f869782af96  yb::ql::PTSelectStmt::Analyze()
    @     0x7f869782a2c3  yb::ql::PTSelectStmt::AnalyzeIndexes()
    @     0x7f869782af5b  yb::ql::PTSelectStmt::Analyze()
    @     0x7f86977d25fd  yb::ql::ParseTree::Analyze()
    @     0x7f86980a7b08  yb::ql::Analyzer::Analyze()
    @     0x7f86982b93f7  yb::ql::QLProcessor::Analyze()
    @     0x7f86982b95ae  yb::ql::QLProcessor::Prepare()
    @     0x7f86982b6e1f  yb::ql::Statement::Prepare()
    @     0x7f8698f9430c  yb::cqlserver::CQLProcessor::ProcessRequest()
    @     0x7f8698f962c5  yb::cqlserver::CQLProcessor::ProcessRequest()
    @     0x7f8698f96631  yb::cqlserver::CQLProcessor::ProcessCall()
    @     0x7f8698fb31ef  yb::cqlserver::CQLServiceImpl::Handle()
    @     0x7f868fb8e75f  yb::rpc::ServicePoolImpl::Handle()
    @     0x7f868fb3246e  yb::rpc::InboundCall::InboundCallTask::Run()
    @     0x7f868fb9bbea  yb::rpc::(anonymous namespace)::Worker::Execute()
    @     0x7f868e4f9b7d  yb::Thread::SuperviseThread()
    @     0x7f8688ea16db  start_thread
```

**Comment by nocaway** [comment]
The issue can be reproduced with the following test cases.

Case 1
```
ycqlsh:test> create table tab3(i int, j int, k int, l int, primary key (i, k)) with transactions = { 'enabled' : true };
ycqlsh:test> create index tab3_idx on tab3(k);
ycqlsh:test> select k from tab3 where k = 1 order by j;
NoHostAvailable: 
```

Case 2
```
ycqlsh> create schema test;
ycqlsh> use test;
ycqlsh:test> create table tab(i int, j jsonb, k int, primary key (i, k)) with transactions = { 'enabled' : true };
ycqlsh:test> create index tab_idx on tab(k, j->>'x');
ycqlsh:test> select * from tab where k = 1 order by j->>'y';
InvalidRequest: Error from server: code=2200 [Invalid query] message="Invalid Arguments. Order By clause contains invalid expression
select * from tab where k = 1 order by j->>'y';
                                       ^^^^^^^
 (ql error -304)"
ycqlsh:test> select * from tab where k = 1 order by j;
NoHostAvailable: 
```

**Comment by nocaway** [comment]
The correct behavior is to raise error when ORDER BY is used on columns that are not part of the index key.
- CQL uses the hash / partition columns in WHERE clause to figure out which index should be used for scanning.
- CQL requires that the ORDER BY clause can be used only on clustering columns of that index. Therefore, an error should be raised in this case when order-by column is not one of the clustering columns.

**Comment by nocaway** [comment]
This issue has been fixed in master branch and backported to 2.1 and 2.2 branches.
It'll be available for next releases on these branches.
